File: Projects/codechef/SMRSTR.java
package codechef;

import java.util.Scanner;

class SMRSTR {
    public static void main(String[] args) throws java.lang.Exception {
        Scanner stdin = new Scanner(System.in);
        int t = stdin.nextInt();
        while (t-- > 0) {
            int n = stdin.nextInt();
            int q = stdin.nextInt();
            long p = 1;
            while (n-- > 0 && p <= (int) 1e9)
                p *= stdin.nextInt();
            stdin.nextLine();
            while (q-- > 0)
                System.out.printf("%d ", stdin.nextInt() / p);
            System.out.println();
        }
    }
}

File: Projects/codechef/SMRSTR.kt
package codechef

import java.io.IOException
import java.util.*

internal object SMRSTR {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        var t = stdin.nextInt()
        while (t-- > 0) {
            var n = stdin.nextInt()
            var q = stdin.nextInt()
            var p: Long = 1
            while (n-- > 0 && p <= 1e9.toInt()) p *= stdin.nextInt().toLong()
            stdin.nextLine()
            while (q-- > 0) System.out.printf("%d ", stdin.nextInt() / p)
            println()
        }
        stdin.close()
    }
}